TeamCode/src/main/java/org/firstinspires/ftc/teamcode/autoFacingDepot.java

```diff
@@ -0,0 +1,326 @@
+package org.firstinspires.ftc.teamcode;
+
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.util.ElapsedTime;
+import com.sun.tools.javac.util.Position;
+import java.util.List;
+import org.firstinspires.ftc.robotcore.external.ClassFactory;
+import org.firstinspires.ftc.robotcore.external.navigation.VuforiaLocalizer;
+import org.firstinspires.ftc.robotcore.external.navigation.VuforiaLocalizer.CameraDirection;
+import org.firstinspires.ftc.robotcore.external.tfod.TFObjectDetector;
+import org.firstinspires.ftc.robotcore.external.tfod.Recognition;
+    //I think I know why this is breaking
+
+
+@Autonomous(name = "Autonomous One", group = "Things")
+// @Disabled
+public class autoFacingDepot extends LinearOpMode {
+
+    DcMotor leftDrive;
+    DcMotor rightDrive;
+
+    private static final String TFOD_MODEL_ASSET = "RoverRuckus.tflite";
+    private static final String LABEL_GOLD_MINERAL = "Gold Mineral";
+    private static final String LABEL_SILVER_MINERAL = "Silver Mineral";
+
+    private static final String VUFORIA_KEY = "Acq/trz/////AAAAGXUOe/SqwE6QmoOrIXJe3S4va9eMDKZHt3tVhm6SIswWHx9Je8DgPAhpIK/mvU8vKs30ybIcUOPqYu738fQl/zQe0DKmzR0SxpqcpnH2VIiteHk8vjfmCuZQGX0PWcaJ/rgat+Fm999sc4UgPuyRo86DpJZ73ZVSZ7zpvyQgH5xkrj42cgFpcfPeMuQGgrqxf4+LT8FXV0NlLbXJzCIbniMEvyHiWd/YMbAO2x83oXa6bjZBUjZlCCUN+EhuKKQlCdfwxzN0aqQEHy8U1svpOPGd7SDp5FmIZsVdt089IGrcwDPYqkgg61sFL7PgaVQAffT6WeUrZq0xXg78FH5i1VYsVkETDu/I69lOdQXKjGlY";
+
+    private VuforiaLocalizer vuforia;
+
+    private TFObjectDetector tfod;
+
+    private MineralPosition position;
+
+    enum MineralPosition {
+
+        CENTER, LEFT, RIGHT
+
+    }
+
+    public void resetEncoders() {
+        leftDrive.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        rightDrive.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        leftDrive.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        rightDrive.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+    }
+
+    int targetPosition = 0;
+    double power = 0;
+
+    int leftPosition = leftDrive.getCurrentPosition();
+    int rightPosition = rightDrive.getCurrentPosition();
+
+    public void waitToFinish() {
+        while (opModeIsActive() && (leftDrive.isBusy() || rightDrive.isBusy())) {
+            telemetry.addData("Left Encoder Position", leftPosition);
+            telemetry.addData("Right Encoder Position", rightPosition);
+            telemetry.update();
+            idle();
+        }
+    }
+
+    public void rotateLeft() {
```

Comment:
Your three "action" functions, rotateLeft, rotateRight, and driveForward, all require a setup of your power and targetPosition properties.  I think it would be cleaner if these three functions too both of those values as parameters, instead of requiring proper setup beforehand.  Better encapsulation.